spec/policies/form_policy_spec.rb

```diff
@@ -146,105 +150,54 @@
-      }
-    end
-
-    let(:gds_forms) { build_list :form, 2, organisation_id: 1 }
-
-    it "uses organisation id to scope what forms a user can see" do
-      organisation_id = instance_double(Integer, "organisation_id")
-      scope = class_spy(Form)
-
-      allow(user).to receive_message_chain(:organisation, :id) { organisation_id } # rubocop:disable RSpec/MessageChain
-
-      described_class.new(user, scope).resolve
-
-      expect(scope).to have_received(:where).with(organisation_id:)
-    end
-
-    context "with a trial user role" do
-      let(:user) { build :user, :with_trial_role, id: 123 }
-      let(:form) { build(:form, creator_id: 123, organisation_id: nil) }
-
-      before do
-        ActiveResource::HttpMock.respond_to do |mock|
-          mock.get "/api/v1/forms?creator_id=123", headers, [form].to_json, 200
+    let(:policy_scope) { described_class.new(user, scope) }
+    let(:scope) { class_double(Form) }
+    let(:user) { build(:user, :with_trial_role) }
+
+    describe "#resolve" do
+      context "when user is on trial" do
+        before do
+          allow(scope).to receive(:where).with(creator_id: user.id)
         end
-      end
 
-      it "only shows forms the user created" do
-        expect(policy_scope.resolve).to eq [form]
-      end
-    end
-
-    context "with a non-trial user role" do
-      let(:organisation) { build :organisation, id: 1, slug: "test-org" }
-      let(:form) { build(:form, organisation_id: 1, creator_id: 1234) }
-      let(:user) { build :user, role: :editor, organisation:, id: 123 }
-
-      before do
-        ActiveResource::HttpMock.respond_to do |mock|
-          mock.get "/api/v1/forms?organisation_id=1", headers, [form].to_json, 200
+        it "returns only their records" do
+          policy_scope.resolve
+          expect(scope).to have_received(:where).with(creator_id: user.id)
         end
       end
 
-      it "only shows forms belonging to the user's organisation" do
-        expect(policy_scope.resolve).to eq [form]
-      end
-    end
-
-    context "with no organisation set" do
-      context "with editor role" do
-        let(:user) { build :user, :with_no_org, role: :editor }
-
-        it "raises an error" do
-          expect { policy }.to raise_error FormPolicy::UserMissingOrganisationError
-        end
-      end
+      (User.roles.keys - %w[trial]).each do |role|
+        context "when user is not a trial user but a #{role}" do
+          let(:user) { build(:user, role:) }
 
-      context "with trial role" do
-        let(:user) { build :user, :with_trial_role }
+          before do
+            allow(scope).to receive(:where).with(organisation_id: user.organisation.id)
+          end
 
-        it "does not an error" do
-          expect { policy }.not_to raise_error
+          it "returns only their organisation records" do
+            policy_scope.resolve
+            expect(scope).to have_received(:where).with(organisation_id: user.organisation.id)
```

Comment:
Testing against the methods called on a double of Form feels less realistic than than using the HTTP mock.

I think chopping off the scope of the tests at this point is better though.

The details of how the `Form` class fetches objects shouldn't be part of these tests. We were mocking the return values to those calls anyway. They weren't testing anything more solid as we were supplying the answers.